src/modules/lost/functions.c

```diff
@@ -258,17 +282,95 @@ int lost_held_function(struct sip_msg *_m, char *_con, char *_pidf, char *_url,
 	LM_DBG("held location request: [%s]\n", que.s);
 
 	/* send locationRequest to location server - HTTP POST */
-	curlres = httpapi.http_connect(_m, &con, NULL, &res, mtheld, &que);
+	if(con.s != NULL && con.len > 0) {
+
+		LM_DBG("using connection [%.*s]\n", con.len, con.s);
+
+		/* send via connection */
+		curl = httpapi.http_connect(_m, &con, NULL, &res, mtheld, &que);
+	} else {
+		/* we have no connection ... do a NAPTR lookup */
+		if(lost_parse_host(did.s, &host, &flag) > 0) {
+
+			LM_DBG("no conn. trying NATPR lookup [%.*s]\n", host.len, host.s);
+
+			/* remove '[' and ']' from string (IPv6) */
+			if(flag == AF_INET6) {
+				host.s++;
+				host.len = host.len - 2;
+			}
+			/* is it a name or ip ... check nameinfo (reverse lookup) */
+			len = 0;
+			ipstr = lost_copy_string(host, &len);
```

Comment:
Its necessary to make a copy here, can not the host.s be used?

Reply:
the `inet_..()` function called in a next step requires a zero terminated string